### ISSUE TITLE
Design Intent - Accessibility Feedback - BAH-MBS-SelfServ, Pre-Need Self Service, Pre-Need Address Validation

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#81625]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- 

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Must:** As you refine this design, be careful and deliberate about how the form inputs are structured -- and what that means for assistive technology users.
   
   Two big assistive tech use cases to be thinking about: screen reader users and voice command/speech recognition users. 
   
   For screen reader users, the typical interaction pattern as they move through a form is to press [tab] to move through the interactive elements of the page (links, buttons, form inputs) rather than having their screen reader announce everything start to finish. That means text that falls outside of interactive elements will likely be missed, unless it's programmatically associated with an appropriate interactive element.
   
   For (sighted) voice users, the typical interaction pattern is to say "Click [name of element]" and expect that element to be clicked, where [name of element] is the text they see on the screen. That means the elements they want to click on need to have distinct, unambiguous, and pronounceable names, and those names need to be visibly associated with the right element.
   
   In the mockups you shared, I think these interactions could break down when [asking users to choose between the address they entered and the suggested address](https://www.figma.com/proto/6ZBtsZJt7Llx1jA5rR8T83/Pre-Need---Address-Validation-Prototype?page-id=94%3A75077&type=design&node-id=785-140520&viewport=4446%2C10397%2C0.64&t=ZccDh6NTXbP8EUMn-1&scaling=min-zoom&starting-point-node-id=785%3A139239&show-proto-sidebar=1).
   - Screen reader users are likely to tab right over the warning alert, and won't hear the "We can't confirm..." and "Tell us which..." text.
   - The "You entered" and "Suggested address" text doesn't look like it's part of the radio button labels, so it won't be announced when tabbing to each radio button. Instead, you'll just hear addresses with no context.
   - For a voice user speaking their command, "Click 123 Main St" and "Click 123 Maine Street" are audibly identical.
   
   Thinking about how the HTML behind the page, I would expect something roughly like:
   
   ```
   <fieldset>
      <legend>
         We can't confirm the address you entered with the U.S. Postal Service. Tell us which of these addresses you'd like us to use.
      <legend>
      <input type="radio" id="radio-1" name="radio-1" value="entered" /> <label for="radio-1">Address you entered: 123 Main St, San Diego, California, 91911</label>
      <input type="radio" id="radio-2" name="radio-2" value="suggested" /> <label for="radio-2">Suggested address: 123 Maine St, San Diego, California, 91911</label>
   </fieldset>
   ```
   
   Putting the prompt ("We can't confirm...") in the `legend` of the `fieldset` ensures that it's announced by screen readers when you tab into the `fieldset` for the first time. Putting the type of address ("Address you entered"/"Suggested address") in the `label` ensures that information is announced when tabbing to each of the radio inputs. And putting the type of address in the `label` also makes those labels distinct from each other when spoken out loud.
   
   Note that it may be possible to make a page with this markup and also the visual design shared in Figma -- but it may not be able to do that using VA design system components. I highly recommend using VA design system components whenever possible, which may mean you'll need to make some adjustments to the visual design to accommodate them and these accessibility requirements.
- [x] **Engineering note:** There was some conversation during the meeting about whether it made sense to the user flow to take you back to the mailing address page after you've confirmed your address. If you decide that this is the appropriate flow, then when returning to the mailing address page, make sure keyboard focus is set to the success alert using `tabindex="-1"`.
   
   Putting keyboard focus on the success alert ensures that the alert will be the first thing announced by screen readers when that page loads. Screen reader users won't typically have the immediate visual feedback that the change they made is complete. Giving them that immediate announcement saves them from having to hunt down the right input to verify that any changes made were actually made.
- [x] **Must:** As you refine this design further, make sure the success alert text matches the action taken by the user. A user that kept the address they entered shouldn't get a "Mailing address updated" alert, since they didn't update their mailing address.
- [x] **Should:** For the situation where [no recommendations are available](https://www.figma.com/proto/6ZBtsZJt7Llx1jA5rR8T83/Pre-Need---Address-Validation-Prototype?page-id=94%3A75077&type=design&node-id=1024-19262&viewport=4446%2C10397%2C0.64&t=ZccDh6NTXbP8EUMn-1&scaling=min-zoom&starting-point-node-id=785%3A139239&show-proto-sidebar=1), it seems odd to offer "No recommendations available" as an option for the user to select. It's not clear what selecting that option would do that's different from what would happen if they select the address they entered. I'd recommend exploring whether this step of the flow is necessary in this case, or if the information can be communicated in a purely informational way rather than requiring the user to make a selection.
- [x] **Should:** I'd encourage you to move phone and email address out of this interaction and onto their own page. In general we try to follow a [one thing per page approach](https://design.va.gov/patterns/ask-users-for/a-single-response), which reduces cognitive load and makes each individual interaction a lot simpler. In this case, I think it breaks the mental model a bit to ask a user for a mailing address, then ask for phone and email, then make them answer a question about their mailing address again. Focusing this interaction on just the mailing address keeps the user focused on that one thing, rather than making them switch between different types of contact information.
   
   

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone
- [ ] Add the Collaboration Cycle initiative epic

**Comment by barron-charles** [comment]
Hey Brian we have some updates with what we have so far. Let me know if this type of design suites what you were thinking from the comments (photo attached). 

1. Radio button screenshot attached for reference. Let me know your thoughts if this is what you were thinking in terms of the write out you provided.
2. We delete the success alert entirely.
3. We are following with with the design team on this one, but the initial idea is to have the second field either removed or default from the entered address.
4. In terms of moving the phone number and email, I have noted it as a potential enhancement to look into with the team.

**Comment by barron-charles** [comment]
@briandeconinck if you don't mind confirming #1 is what you were looking for above.

**Comment by briandeconinck** [comment]
@barron-charles Yes! This looks great, thank you!

Also, sorry for the delayed response. I'm tagged on a ton of GitHub tickets and sometimes miss some notifications. Please don't hesitate to ping me in Slack if I haven't replied promptly. 🙂

**Comment by barron-charles** [comment]
Thank you and appreciate the clarification 🙂.
